CMakeLists.txt

```diff
@@ -93,7 +99,8 @@ endif()
 
 target_compile_options(yaml-cpp
   PRIVATE
-    $<${not-msvc}:-Wall -Wextra -Wshadow -Weffc++ -Wno-long-long>
+    $<${not-msvc}:-Wall -Wextra -Wshadow -Wno-long-long>
```

Comment:
This looks separate from the primary point of the PR?

Reply:
it is, but adding the `YAML_CPP_MASTER_PROJECT` option from this PR makes this fix very easy.  It's a partial fix for issue #970 (basically if you're not the toplevel project omit the -Weffc++ option.)  If you'd rather once the other change is approve I can make a new PR for this.

Reply:
I guess I need to add the `not-msvc` to the check as well (the option would be erroneously added on msvc when then top-level project) ...may be just as easy to omit all these options if this library is not the top-level project.